src/content/advanced/gitops/environments/index.md

```diff
@@ -0,0 +1,386 @@
+weight: 70
+menu:
+  main:
+    parent: advanced-gitops
+    identifier: advanced-gitops-environments
+user_questions:
+  - How can I manage different environments with GitOps?
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-honeybadger
+last_review_date: 2023-02-10
+---
+
+# Add Workload Cluster environments
+
+You might want to set up multiple, similar Workload Clusters that serve as for example development, staging and production environments. You can utilize `bases` to achieve that. Let's take a look at the `/bases/environments` folder structure.
+
+## General note
+
+It is possible to solve the environment propagation problem in multiple ways, notably by:
+
+- using a multi-directory structure, where each environment is represented as a directory in a `main` branch of a single
+  repository
+- using a multi-branch approach, where each branch corresponds to one environment, but they are in the same repo
+- using a multi-repo setup, where there's one root repository providing all the necessary templates, and then there is another repository per environment.
+
+Each of these approaches has pros and cons. Giant Swarm relies on the multi-directory approach for multiple reasons. It is a single repo and single branch which serves as the source of truth for all the environments. Easy for template sharing and a relatively simple way to compare and promote configuration across environments. On the other hand, it needs extra work for access control management, template versioning or environment drift detection.
+
+## Environments
+
+We are creating a new `base` type called environment. Inside there will be two folders.
+
+The `stages` folder is how we propose to group environment specifications. There is a good reason for this additional layer. You can use it for having multiple different clusters - like the dev, staging and production - but also to have multiple different regions where you want to spin these clusters up.
+
+The `regions` folder is how we propose to group specifications for configurations referred to different locations or regions in the different cloud providers or datacenters.
+
+__Note__: If you want to use multiple environment templates to create a single cluster that uses `App CR`s for deployments, for example, you would like to use `dev` layout to set app configuration and then use `east` from the `data-centers` to set the IP ranges, you might run into issues around merging configurations, as currently one configuration source (i.e. `ConfigMap` in `spec.config.configMap`) completely overrides the whole value of the same attribute coming from the other base. We're working to remove this limitation. This means for example that setting `machinePools` in several files will result on selecting only the full block of the file with higher priority.
+
+In order to avoid config collisions, we are specifying different config changes in different environment folders. In our case:
```

Comment:
```suggestion
In order to avoid possible collisions in the configuration, we define the different changes in the configuration across several environment folders. In our case:
```